Fix llama quantize_per_token numerics

andrewor14 opened the pull request:
Summary:
The existing implementation can produce quantized values
outside the quant range, since we add the zero points after
clamping. This was not a problem for symmetric quantization
since zero points are 0 there, but causes dqlinear numerics
to diverge significantly from the lowered implementation for
asymmetric quantization.

Reviewed By: digantdesai

Differential Revision: D54320424

digantdesai reviewed (approved):
Make sure CI is clean before landing

andrewor14 committed:
Fix llama quantize_per_token numerics

Summary:
The existing implementation can produce quantized values
outside the quant range, since we add the zero points after
clamping. This was not a problem for symmetric quantization
since zero points are 0 there, but causes dqlinear numerics
to diverge significantly from the lowered implementation for
asymmetric quantization.

Reviewed By: digantdesai

Differential Revision: D54320424
